cmd/dep/ensure.go

```diff
@@ -215,23 +213,23 @@ func (cmd *ensureCommand) Run(ctx *ctx, args []string) error {
 		return errors.Wrap(err, "ensure Solve()")
 	}
 
-	sw := safeWriter{
-		root: p.absroot,
-		m:    p.m,
-		l:    p.l,
-		nl:   solution,
-		sm:   sm,
+	sw := dep.SafeWriter{
+		Root:          p.AbsRoot,
+		Manifest:      p.Manifest,
+		Lock:          p.Lock,
+		NewLock:       solution,
+		SourceManager: sm,
 	}
 
 	// check if vendor exists, because if the locks are the same but
 	// vendor does not exist we should write vendor
 	var writeV bool
-	vendorExists, _ := isDir(filepath.Join(sw.root, "vendor"))
+	vendorExists, _ := dep.IsDir(filepath.Join(p.AbsRoot, "vendor"))
```

Comment:
Nice catch! My first attempt was to not make `SafeWriter` fields public, and since `p.AbsRoot` was the same, I was using that. But `SafeWriter` fields are public now anyway so it makes sense to change it back and forgot about it. Will fix that right now